integration/clean/dedicated_host/clean.go

```diff
@@ -1,54 +1,45 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: MIT
 
-//go:build clean
-// +build clean
-
-package main
+package dedicated_host
 
 import (
 	"context"
+	"log"
+	"time"
+
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
 	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
-	"log"
-	"time"
 )
 
 // Can't release a host if it was being used within the last 24 hr add 2 hr as a buffer
-const keepDurationDedicatedHost = -1 * time.Hour * 26
-const tagName = "tag:Name"
-const tagValue = "IntegrationTestMacDedicatedHost"
-
-func main() {
-	err := cleanDedicatedHost()
-	if err != nil {
-		log.Fatalf("errors cleaning %v", err)
-	}
-}
+const (
+	Type     = "dedicated_host"
+	tagName  = "tag:Name"
+	tagValue = "IntegrationTestMacDedicatedHost"
+)
 
-func cleanDedicatedHost() error {
+func Clean(ctx context.Context, expirationDate time.Time) error {
```

Comment:
A point of code cleanliness, I think, would be to define an interface for resource cleaners and have each of these different cleaners implement the interface, though that might be overkill.

```go
type Cleaner interface {
    Clean(ctx context.Context, expirationDate time.Time) error
    Type() string
}
```

Reply:
Although it would look elegant, I don't think this is worthwhile. I'm still leaving it here though.

Reply:
I thought of it but agree on its overkill a little bit and we don't maintain this much.